src/main/java/uk/gov/ida/matchingserviceadapter/resources/UnknownUserAttributeQueryResource.java

```diff
@@ -30,45 +44,71 @@
 
     private static final Logger LOG = LoggerFactory.getLogger(UnknownUserAttributeQueryResource.class);
 
-    private final DocumentToInboundMatchingServiceRequestMapper documentToInboundMatchingServiceRequestMapper;
     private final Function<OutboundResponseFromUnknownUserCreationService, Element> responseElementTransformer;
     private final SoapMessageManager soapMessageManager;
-    private final UnknownUserAttributeQueryHandler attributeQueryHandler;
+    private final UnknownUserResponseGenerator attributeQueryHandler;
+    private final ElementToOpenSamlXMLObjectTransformer<AttributeQuery> attributeQueryUnmarshaller;
+    private final AttributeQueryService attributeQueryService;
+    private final MatchingServiceProxy matchingServiceProxy;
+    private final AssertionDecrypter assertionDecrypter;
 
     @Inject
     public UnknownUserAttributeQueryResource(
-            DocumentToInboundMatchingServiceRequestMapper documentToInboundMatchingServiceRequestMapper,
             Function<OutboundResponseFromUnknownUserCreationService, Element> responseElementTransformer,
             SoapMessageManager soapMessageManager,
-            UnknownUserAttributeQueryHandler attributeQueryHandler) {
-        this.documentToInboundMatchingServiceRequestMapper = documentToInboundMatchingServiceRequestMapper;
+            UnknownUserResponseGenerator attributeQueryHandler,
+            ElementToOpenSamlXMLObjectTransformer<AttributeQuery> attributeQueryUnmarshaller,
+            AttributeQueryService attributeQueryService,
+            MatchingServiceProxy matchingServiceProxy,
+            AssertionDecrypter assertionDecrypter) {
         this.responseElementTransformer = responseElementTransformer;
         this.soapMessageManager = soapMessageManager;
         this.attributeQueryHandler = attributeQueryHandler;
+        this.attributeQueryUnmarshaller = attributeQueryUnmarshaller;
+        this.attributeQueryService = attributeQueryService;
+        this.matchingServiceProxy = matchingServiceProxy;
+        this.assertionDecrypter = assertionDecrypter;
     }
 
     @POST
     @Timed(name= Urls.SOAP_TIMED_GROUP)
     public Response receiveUnknownUserRequest(Document attributeQueryDocument) {
+        AttributeQuery attributeQuery = unwrapAttributeQuery(attributeQueryDocument);
+        List<Assertion> assertions = assertionDecrypter.decryptAssertions(new EncryptedAssertionContainer(attributeQuery));
```

Comment:
Same comment as above here - the `decryptAssertions` method was only supposed to be called with assertions that came from a validated source, and here you're circumventing that. We should either change `decryptedAssertions` to be less strict or validate the AQR before we try to decrypt its assertions.